crawler/http/crawler/spiders/sycamore.py

```diff
@@ -4,24 +4,31 @@
 import re
 import scrapy
 import time
-import urllib.parse
+import urllib
 
 
 class SycamoreSpider(scrapy.Spider):
     name = "sycamore"
 
     def __init__(self, category=None, *args, **kwargs):
         self.dest_dir = kwargs.get("dest_dir", ".scrapy/downloads")
+        self.prefix = None
         if "preset" in kwargs:
             self._setup_by_preset(kwargs["preset"])
-        elif "domain" in kwargs and "url" in kwargs:
-            self.allowed_domains = [kwargs["domain"]]
+        elif "url" in kwargs:
             self.start_urls = [kwargs["url"]]
-        elif "domain" in kwargs or "url" in kwargs:
-            raise RuntimeError("Should have both -a domain=... and -a url=...")
+            if "prefix" in kwargs:
+                self.prefix = kwargs["prefix"]
+            if "domain" in kwargs:
+                self.allowed_domains = [kwargs["domain"]]
+            else:
+                u = urllib.parse.urlparse(kwargs["url"])
+                d = u.hostname.removeprefix("www.")
```

Comment:
Does this still work if the url is given like "https://www.aryn.ai"?

Reply:
Since that's exactly the string I used to test it, yes.  It downloaded from docs.aryn.ai and blog.aryn.ai 

Why would you think it wouldn't?